src/group_theory/subgroup.lean

```diff
@@ -404,13 +406,11 @@ end
 @[to_additive] instance fintype_bot : fintype (⊥ : subgroup G) := ⟨{1},
 by {rintro ⟨x, ⟨hx⟩⟩, exact finset.mem_singleton_self _}⟩
 
-@[simp] lemma _root_.add_subgroup.card_bot : fintype.card (⊥ : add_subgroup A) = 1 :=
-fintype.card_eq_one_iff.2
-  ⟨⟨(0 : A), set.mem_singleton 0⟩, λ ⟨y, hy⟩, subtype.eq $ add_subgroup.mem_bot.1 hy⟩
-
--- `@[to_additive]` doesn't work, because it converts the `1 : ℕ` to `0`.
-@[simp] lemma card_bot : fintype.card (⊥ : subgroup G) = 1 :=
-fintype.card_eq_one_iff.2
+/-! curly brackets `{}` are used here instead of instance brackets `[]` because
+  the instance in a goal is often not the same as the one inferred by type class inference.  -/
+@[simp, to_additive] lemma card_bot {h : fintype.{u} (⊥ : subgroup G)} :
+  fintype.card (⊥ : subgroup G)  = 1 :=
+(@fintype.card_eq_one_iff (↥(⊥ : subgroup G)) h).2
```

Comment:
Repeating https://github.com/leanprover-community/mathlib/pull/8383#discussion_r674375932 since the diff moved around
```suggestion
@[simp, to_additive] lemma card_bot {_ : fintype ↥(⊥ : subgroup G)} :
  fintype.card (⊥ : subgroup G) = 1 :=
fintype.card_eq_one_iff.2
```

This worked just fine when I tested it on your last revision - does it not work for you?

Reply:
Sorry, I forgot about that comment